src/test/java/com/exasol/releasedroid/repository/MavenPomParserTest.java

```diff
@@ -47,38 +44,41 @@ void testParseMavenPomWithPlugins() throws IOException {
         final String pom = "<project>" //
                 + "    <artifactId>my-test-project</artifactId>" //
                 + "    <version>1.2.3</version>" //
+                + "    <properties>" //
+                + "        <plugin.version>5.0.4</plugin.version>" //
+                + "    </properties>" //
                 + "    <build>" //
                 + "        <plugins>" //
                 + "            <plugin>" //
                 + "                <artifactId>some-plugin</artifactId>" //
+                + "                <version>0.5.0</version>" //
                 + "            </plugin>" //
                 + "            <plugin>" //
                 + "                <artifactId>maven-assembly-plugin</artifactId>" //
-                + "                 <configuration>" //
-                + "                    <finalName>virtual-schema-dist-${vscjdbc.version}-bundle-${version}</finalName>"
-                + "                </configuration>" //
+                + "                <version>1.0.3</version>" //
                 + "            </plugin>" //
                 + "            <plugin>" //
                 + "                <artifactId>some-other-plugin</artifactId>" //
-                + "                 <configuration>" //
+                + "                <version>${plugin.version}</version>" //
+                + "                <configuration>" //
                 + "                </configuration>" //
                 + "            </plugin>" //
                 + "        </plugins>" //
                 + "    </build>" //
                 + "</project>";
         final MavenPom mavenPom = getMavenPom(pom);
-        final List<MavenPlugin> plugins = mavenPom.getPlugins();
+        final Map<String, MavenPlugin> plugins = mavenPom.getPlugins();
         assertAll(() -> assertThat(mavenPom.getVersion(), equalTo("1.2.3")), //
                 () -> assertThat(mavenPom.getArtifactId(), equalTo("my-test-project")), //
-                () -> assertThat(mavenPom.hasProperties(), equalTo(false)), //
+                () -> assertThat(mavenPom.hasProperties(), equalTo(true)), //
                 () -> assertThat(mavenPom.hasPlugins(), equalTo(true)), //
                 () -> assertThat(plugins.size(), equalTo(3)), //
-                () -> assertThat(plugins.get(0).getArtifactId(), equalTo("some-plugin")), //
-                () -> assertThat(plugins.get(0).hasConfiguration(), equalTo(false)), //
-                () -> assertThat(plugins.get(1).getArtifactId(), equalTo("maven-assembly-plugin")), //
-                () -> assertThat(plugins.get(1).hasConfiguration(), equalTo(true)), //
-                () -> assertThat(plugins.get(2).getArtifactId(), equalTo("some-other-plugin")), //
-                () -> assertThat(plugins.get(2).hasConfiguration(), equalTo(true)) //
+                () -> assertThat(plugins.containsKey("some-plugin"), equalTo(true)), //
```

Comment:
maybe you could also split the test into multiple small ones. so for example one for plugins and one for version and so on. By that you would only need to define the required parts in the pom for each test.
That would maybe give a bit better feedback what exactly failed if something is broken.

But maybe it's also not necessary here.

Reply:
The test checks that all reports are finally merged into one and I've not missed anything. If I only test small methods, there is a chance that I forgot to merge one of them (I just had this problem today because my test only tested small methods)